### PR TITLE
Add an elementwise absolute value function

### PR DESCRIPTION
**Pull request by okuta**
This PR is implementation of #11.

### CODE REVIEW

**Comment by beam2d** [comment]
I'll check it.

File: chainer/functions/basic_math.py [hunk 1]
```diff
@@ -26,6 +26,27 @@ def neg(x):  # -x
...
+    def backward_gpu(self, x, gy):
+        gx0 = cuda.empty_like(x[0])
+        cuda.elementwise(
+            'float* gx0, const float* x0, const float* gy',
+            'gx0[i] = ((x0[i] > 0) - (x0[i] < 0)) * gy[i]',
+            'back_abs')(gx0, x[0], gy[0])
+        return gx0,
+
+
+def abs(x):
```
Comment (beam2d) [inline chainer/functions/basic_math.py]: Use a different name to stop hiding the built-in function.

File: chainer/functions/basic_math.py [hunk 2]
```diff
@@ -26,6 +26,27 @@ def neg(x):  # -x
     return Neg()(x)
 
 
+class Abs(function.Function):
+
+    def forward(self, x):
+        return utils.force_array(x[0].__abs__()),
```
Comment (beam2d) [inline chainer/functions/basic_math.py]: Use the built-in abs function (see also the comment below).

File: chainer/functions/basic_math.py [hunk 3]
```diff
@@ -26,6 +26,27 @@ def neg(x):  # -x
...
+
+    def backward_cpu(self, x, gy):
+        return utils.force_array(numpy.sign(x[0]) * gy[0]),
+
+    def backward_gpu(self, x, gy):
+        gx0 = cuda.empty_like(x[0])
+        cuda.elementwise(
+            'float* gx0, const float* x0, const float* gy',
+            'gx0[i] = ((x0[i] > 0) - (x0[i] < 0)) * gy[i]',
+            'back_abs')(gx0, x[0], gy[0])
```
Comment (beam2d) [inline chainer/functions/basic_math.py]: Use a consistent kernel name, e.g. `abs_bwd`, following backprop kernels of other functions.

**Comment by beam2d** [comment]
LGTM except some comments.

**Comment by okuta** [comment]
Thank you for your comments. I will fix them.

**Comment by beam2d** [comment]
Thank you!

**Pull request merged**